lib/src/generators/openapi3/__spec-examples__/contract-with-schemaprops.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  api,
+  body,
+  endpoint,
+  headers,
+  request,
+  response,
+  String,
+  Integer
+} from "@airtasker/spot";
+
+@api({ name: "contract" })
+class Contract {}
+
+@endpoint({
+  method: "GET",
+  path: "/users"
+})
+class EndpointWithExampleOnHeaders {
+  @request
+  request(
+    @headers
+    headers: {
+      /** property-schemaprop description for string
+       * @schemaprop minLength
+       * 12
+       * @schemaprop maxLength
+       * 20
+       * @schemaprop pattern
+       * "^[0-9a-z_]+$"
+       *  */
+      status: String;
+      /** property-schemaprop description for integer
+       * @schemaprop minimum
+       * 1
+       * @schemaprop maximum
+       * 100
+       * @schemaprop default
+       * 42
```

Comment:
We are currently using `@default` tag to convey default values in [`@oa3serverVariables`](https://github.com/airtasker/spot/wiki/Spot-syntax#openapi-3-servers). I think it makes sense for `@default` to be used generically wherever we need default values.

For any OpenAPI specific tags, the convention we approached this with is to prefix with:

* `@oa*` for OpenAPI 2 & 3
* `@oa2*` for OpenAPI 2
* `@oa3*` for OpenAPI 3

🙇‍♂️ Sorry we hadn't documented this - see https://github.com/airtasker/spot/issues/1050 for more context.

For these schema props we can prefix with `oa*` since they'll be applicable for both OpenAPI 2 & 3.

Reply:
Hello @lfportal,
It's a good idea to rename it `@oaschemaProp` and make `@default` tag works in schema properties as well.

Do you agree?

Reply:
@Thodin3 makes sense, this approach is more practical for now. A larger refactor can be considered later. 👍

Reply:
Yup agreed 👍

Reply:
ok cool